staging/src/k8s.io/kubectl/pkg/cmd/debug/debug.go

```diff
@@ -652,7 +651,7 @@ func waitForContainer(ctx context.Context, podClient corev1client.PodsGetter, ns
 		ev, err := watchtools.UntilWithSync(ctx, lw, &corev1.Pod{}, nil, func(ev watch.Event) (bool, error) {
 			switch ev.Type {
 			case watch.Deleted:
-				return false, errors.NewNotFound(schema.GroupResource{Resource: "pods"}, "")
+				return false, fmt.Errorf("the pod %q can't tolerate the specified node, has been deleted", podName)
```

Comment:
That's assuming only the case from https://github.com/kubernetes/kubernetes/issues/98253, what if the reason for deletion was different though?

Reply:
Thanks @soltysh